bc_obps/registration/models/event.py

```diff
@@ -1,128 +1,25 @@
-from typing import Callable, Dict, Optional, Union
 import uuid
 from django.db import models
-from registration.models import Address, Facility, Operation, TimeStampedModel
-from simple_history.models import HistoricalRecords
-from django.core.exceptions import ValidationError
+from registration.models import TimeStampedModel
 from datetime import datetime
 
 
 class Event(TimeStampedModel):
-    class Types(models.TextChoices):
-        CLOSING_OR_TEMPORARY_SHUTDOWN = "Closing or Temporary Shutdown"
-        ACQUISITION = "Acquisition"
-        TRANSFER_OF_CONTROL = "Transfer of Control"
-        DIVESTMENT = "Divestment"
-        STARTUP = "Startup"
-
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, db_comment="Primary key to identify the event")
-    operation = models.ForeignKey(Operation, null=True, blank=True, on_delete=models.DO_NOTHING, related_name="events")
-    facility = models.ForeignKey(Facility, null=True, blank=True, on_delete=models.DO_NOTHING, related_name="events")
```

Comment:
Since we are using `operation` as a foreign key and `facilities` as a many-to-many field on each child data model, would it make sense to keep those two fields on the abstract data model?

Reply:
@Sepehr-Sobhani  that's what I wanted to do, but I couldn't figure out how to keep the `operation` and `facilities` fields in the `Event` model but then set the `related_name` in each of the `Restart`, `Closure`, `Transfer`, `TemporaryShutdown` so that they have unique names. Any suggestions?

Reply:
You can use something like this: `related_name='%(class)ss'`

Reply:
doc: https://docs.djangoproject.com/en/5.0/topics/db/models/#be-careful-with-related-name-and-related-query-name:~:text=%27%25(class)s%27%20is%20replaced%20by%20the%20lowercased%20name%20of%20the%20child%20class%20that%20the%20field%20is%20used%20in.

Reply:
yet another failure on my part to RTFM. Thanks Sepehr!!!! Will implement this

Reply:
Thank you for doing the hard work. Since you are flexing tomorrow, I can handle comments and suggestions if you'd like.

Reply:
I've still got some time left in my work day. I'll address as much as I can before I log off and "resolve" those conversations. If you can address any outstanding changes I'm not able to get to, that would be tremendously appreciated! <3

Reply:
With pleasure! I can take care of the remaining tasks, and you can review and merge the PR on Monday.